core/embed/projects/boardloader/main.c

```diff
@@ -184,14 +184,16 @@ static uint32_t check_sdcard(void) {
                    "BOOTLOADER IMAGE MAXSIZE too large for IMAGE_CHUNK_SIZE");
 
     const uint32_t headers_end_offset = hdr->hdrlen;
-    const uint32_t code_start_offset = IMAGE_CODE_ALIGN(headers_end_offset);
+    const uint32_t padding_end = IMAGE_CODE_ALIGN(headers_end_offset);
 
-    for (uint32_t i = headers_end_offset; i < code_start_offset; i++) {
+    for (uint32_t i = headers_end_offset; i < padding_end; i++) {
       if (((uint8_t *)sdcard_buf)[i] != 0) {
         return 0;
       }
     }
 
+    const uint32_t code_start_offset = IMAGE_CODE_ALIGN(headers_end_offset);
+
```

Comment:
if that's the case, i don't understand this change.

it seems to me that you're (a) still checking explicitly for zero value padding (that's fine i guess) but also `check_single_hash(..., code_start_offset)` which is `IMAGE_CODE_ALIGN(headers_end_offset)`

Reply:
yeah that is a bug https://github.com/trezor/trezor-firmware/pull/4521/commits/40bcbd8da2ed307cad938d43f22a62ee706beb83